### PR TITLE
OSD: Move hardcore notification from toast to OSD

### PR DESCRIPTION
**Pull request by kamfretoz**
### Description of Changes

This PR moves the Hardcore mode notification from toast to the OSD.

Before:
Screenshot_20240724_153930

After:
Screenshot_20240724_155918

### Rationale behind Changes

Makes it more obvious of its existence and makes it easier to read.

### Suggested Testing Steps

See if the notification shows up properly.

#### Steps to test
1. Enable Hardcore Mode
2. Run a game that has no achievements
3. Observe the notification on the top left

### CODE REVIEW

**Comment by lightningterror** [comment]
I prefer it where it is, but that's just me.

**Comment by PolygonSlayer** [comment]
Just chiming in as I was the one mentioning this in the discord.
I think if there was an option that might be good then for people who prefer the old one. For me I think anything that covers the game screen it would be nice if it was either a way to turn it off or move it. In this case you can't turn this off unless you turn RetroAchivements off entirely which is a bit annoying (at least I was unable to find an option for it).

**Comment by kamfretoz** [comment]
It's more of consistencies as well, everything else already uses the top-left OSD notifications. the hardcore mode is the only one that aren't already (there are also related to connectivity to RCheevos server that still uses the toast but i choose to leave it as is because its lower-priority compared to the actual hardcore status notification).

**Review by lightningterror — APPROVED** [review]
Scratch what I said before, I mixed before and after comparison.

**Comment by F0bes** [comment]
Just for consistency, would you be able to change the other achievement messages to use the OSD messages instead of the fullscreen UI?

"Leaderboard download failed"
"Achievements Disconnected"
etc

**Comment by kamfretoz** [comment]
Done

File: pcsx2/Achievements.cpp [hunk 1]
```diff
@@ -2950,7 +2947,8 @@ void Achievements::LeaderboardFetchAllCallback(
 
 	if (result != RC_OK)
 	{
-		ImGuiFullscreen::ShowToast(TRANSLATE("Achievements", "Leaderboard download failed"), error_message);
+		ImGuiFullscreen::AddNotification("leaderboard_dl_fail", Host::OSD_INFO_DURATION, TRANSLATE("Achievements","Leaderboard Download Failed"), error_message, s_game_icon);
+
```
Comment (F0bes) [inline pcsx2/Achievements.cpp]: Trailing tabs

File: pcsx2/Achievements.cpp [hunk 2]
```diff
@@ -2950,7 +2947,8 @@ void Achievements::LeaderboardFetchAllCallback(
 
 	if (result != RC_OK)
 	{
-		ImGuiFullscreen::ShowToast(TRANSLATE("Achievements", "Leaderboard download failed"), error_message);
+		ImGuiFullscreen::AddNotification("leaderboard_dl_fail", Host::OSD_INFO_DURATION, TRANSLATE("Achievements","Leaderboard Download Failed"), error_message, s_game_icon);
```
Comment (F0bes) [inline pcsx2/Achievements.cpp]: Missing a space after `"Achievements",`

File: pcsx2/Achievements.cpp [hunk 3]
```diff
@@ -2931,7 +2928,7 @@ void Achievements::LeaderboardFetchNearbyCallback(
...
```
Comment (F0bes) [inline pcsx2/Achievements.cpp]: Missing a space after `"Achievements",`

File: pcsx2/Achievements.cpp [hunk 4]
```diff
@@ -1357,8 +1355,7 @@ void Achievements::HandleServerReconnectedEvent(const rc_client_event_t* event)
 	MTGS::RunOnGSThread([]() {
 		if (ImGuiManager::InitializeFullscreenUI())
 		{
-			ImGuiFullscreen::ShowToast(TRANSLATE_STR("Achievements", "Achievements Reconnected"),
-				TRANSLATE_STR("Achievements", "All pending unlock requests have completed."), Host::OSD_INFO_DURATION);
+			ImGuiFullscreen::AddNotification("achievements_reconnect", Host::OSD_INFO_DURATION, TRANSLATE_STR("Achievements", "Achievements Reconnected"), TRANSLATE_STR("Achievements", "All pending unlock requests have completed."), s_game_icon);
```
Comment (F0bes) [inline pcsx2/Achievements.cpp:1358]: Can you split these back to two lines, it's really long

File: pcsx2/Achievements.cpp [hunk 5]
```diff
@@ -1343,9 +1343,7 @@ void Achievements::HandleServerDisconnectedEvent(const rc_client_event_t* event)
 	MTGS::RunOnGSThread([]() {
 		if (ImGuiManager::InitializeFullscreenUI())
 		{
-			ImGuiFullscreen::ShowToast(TRANSLATE_STR("Achievements", "Achievements Disconnected"),
-				TRANSLATE_STR("Achievements", "An unlock request could not be completed. We will keep retrying to submit this request."),
-				Host::OSD_ERROR_DURATION);
+			ImGuiFullscreen::AddNotification("achievements_disconnect", Host::OSD_ERROR_DURATION, TRANSLATE_STR("Achievements", "Achievements Disconnected"), TRANSLATE_STR("Achievements", "An unlock request could not be completed. We will keep retrying to submit this request."), s_game_icon);
```
Comment (F0bes) [inline pcsx2/Achievements.cpp:1346]: Same thing, can you split these back to two lines, it's really long

**Review by F0bes — CHANGES_REQUESTED** [review]

**Review by F0bes — APPROVED** [review]
Awesome. lgtm

**Pull request merged**